### ISSUE TITLE
[bug] Editable layout ignored during import()

### ISSUE THREAD

**Issue by jfischer1** [conan-io/conan#5928]
When using editables with different layouts, then copying of files in import() fails. It looks like FileCopier does not take different editable layouts into account.

### Environment Details (include every applicable attribute)
  * Operating system:Win 10
  * Compiler+version: N/A
  * Conan version: 1.19.2
  * Python version: 3.7.4

### Steps to reproduce (Include if Applicable)
1. Create project A that exports a binary file X in folder 'bin'.
2. Make A an editable with a layout where binary folder is 'x64-Debug/bin'.
3. Consume A with project B. Let B try to import X:

```
def imports(self):
    self.copy("X", src="bin", dst="bin", root_package='A')
```

This copy command will not find X. Workaround:
```
def imports(self):
     for bin_dir in self.deps_cpp_info['A'].bindirs:
         self.copy("X", src=bin_dir, dst='bin')
```

### Logs (Executed commands with output) (Include/Attach if Applicable)
not applicable

**Comment by jgsogo** [comment]
Thanks for pointing it out, @jfischer1. Indeed it is not working as expected. 

I've created an unittest to reproduce the error:

```python
# coding=utf-8

import textwrap
import unittest

from conans.model.ref import ConanFileReference
from conans.test.utils.tools import TestClient


class EditableImportTestCase(unittest.TestCase):

    def test_import_from_editable(self):
        # Create a package
        t1 = TestClient()
        req_ref = ConanFileReference.loads("req/version")
        conanfile = textwrap.dedent("""
            from conans import ConanFile

            class Requirement(ConanFile):
                exports_sources = "*"
                def package(self):
                    self.copy("*.lib", dst="lib", src="localpath/lib") 
        """)
        t1.save({"conanfile.py": conanfile,
                 "localpath/lib/file.lib": "whatever"})
        t1.run("create . {}@".format(req_ref))

        # Put the package in editable mode
        layout = textwrap.dedent("""
            [libdirs]
            localpath/lib   ### <--- maybe just "localpath"
        """)
        t1.save({"layout": layout})
        t1.run("editable add -l layout . {}@".format(req_ref))

        # Consume the editable package
        t = TestClient(cache_folder=t1.cache_folder)
        conanfile = textwrap.dedent("""
            from conans import ConanFile

            class App(ConanFile):
                requires = "{}"
                def imports(self):
                    self.output.info("Libdirs: {{}}".format(self.deps_cpp_info['req'].libdirs))
                    self.copy("*.lib", src="lib", dst="lib", root_package="{}")
        """.format(req_ref, req_ref.name))
        t.save({"conanfile.py": conanfile})
        t.run("create . app/version@")

        print(t.out)
        self.assertIn("app/version: Libdirs: ['localpath/lib']", t.out)
        self.assertIn("app/version imports(): Copied 1 '.lib' file: file.lib", t.out)

```

`FileCopier` should iterate over the paths declared in the layout file looking for the files to copy.  

⚠️  Right now I'm not sure if the path in the `libdirs` section of the layout file should be `localpath` or `localpath/lib`. I'm afraid that we will need both: the first one for this copy because of the `src="lib"` while the other one to find the actual library files to link.

**Comment by jgsogo** [comment]
PR https://github.com/conan-io/conan/pull/6208 provides a better alternative for this same problem

**Comment by jgsogo** [comment]
After https://github.com/conan-io/conan/pull/6208 (to be released in Conan v1.22) you can write _symbolic_ directories in the `src` attribute to copy from the folder defined in `cpp_info`:

```
def imports(self):
    self.copy("X", src="@bindirs", dst="bin", root_package='A')
```

This will copy `X` from the folders defined in `cpp_info["bindirs"]` of all the dependencies. Those folders can be the ones defined in the `packagE_info()` method or the ones overridden by a layout file in a package installed as editable.
 

Thanks for your patience!
